crates/pevm/src/mv_memory.rs

```diff
@@ -98,7 +98,7 @@ impl MvMemory {
         let mut last_location_idx = 0;
         while last_location_idx < last_locations.write.len() {
             let prev_location = unsafe { last_locations.write.get_unchecked(last_location_idx) };
-            if !write_set.iter().any(|(l, _)| l == prev_location) {
+            if write_set.iter().all(|(l, _)| l != prev_location) {
```

Comment:
`any` is preferred as it is short-circuiting and this is a common case as most transaction re-execution still writes to the same memory locations. `all` would be visibly slower for transactions that write to many locations.

Reply:
I think this is the semantic equivalent, actually, and there is no performance difference as well. You can write `.all` and `.any` in terms of each other like this:
```rust
fn any(&self, pred: impl FnMut(I) -> bool) -> bool {
  !self.all(|i| !pred(i))
}
fn all(&self, pred: impl FnMut(I) -> bool) -> bool {
  !self.any(|i| !pred(i))
}
```

You can also compare implementations of those 2 in stdlib
https://doc.rust-lang.org/src/core/iter/traits/iterator.rs.html#2751-2756
https://doc.rust-lang.org/src/core/iter/traits/iterator.rs.html#2804-2811

Reply:
Aha, should check the doc before commenting next time. No need to change then 🙏.